conan/tools/cmake/cmakedeps/templates/target_configuration.py

```diff
@@ -60,64 +61,44 @@ def template(self):
         conan_find_apple_frameworks({{ pkg_name }}_FRAMEWORKS_FOUND{{ config_suffix }} "{{ '${' }}{{ pkg_name }}_FRAMEWORKS{{ config_suffix }}}" "{{ '${' }}{{ pkg_name }}_FRAMEWORK_DIRS{{ config_suffix }}}")
 
         # Gather all the libraries that should be linked to the targets (do not touch existing variables)
-        set(_{{ pkg_name }}_DEPENDENCIES{{ config_suffix }} "{{ '${' }}{{ pkg_name }}_FRAMEWORKS_FOUND{{ config_suffix }}} {{ '${' }}{{ pkg_name }}_SYSTEM_LIBS{{ config_suffix }}} {{ deps_targets_names }}")
+        # FIXME: Why is needed to pass all the dependencies to link with the micro-targets?
+        set(_{{ pkg_name }}_DEPENDENCIES{{ config_suffix }} "{{ deps_targets_names }}")
 
         set({{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} "") # Will be filled later
-        set({{ pkg_name }}_LIBRARIES{{ config_suffix }} "") # Will be filled later
         conan_package_library_targets("{{ '${' }}{{ pkg_name }}_LIBS{{ config_suffix }}}"    # libraries
                                       "{{ '${' }}{{ pkg_name }}_LIB_DIRS{{ config_suffix }}}" # package_libdir
                                       "{{ '${' }}_{{ pkg_name }}_DEPENDENCIES{{ config_suffix }}}" # deps
-                                      {{ pkg_name }}_LIBRARIES{{ config_suffix }}   # out_libraries
-                                      {{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }}  # out_libraries_targets
-                                      "{{ config_suffix }}"  # config_suffix
+                                      {{ pkg_name }}_LIBRARIES_TARGETS  # out_libraries_targets
+                                      "{{ config }}" # DEBUG, RELEASE ...
                                       "{{ pkg_name }}")    # package_name
 
+        # The XXXX_LIBRARIES_RELEASE/DEBUG is used for the module (FindXXX.cmake)
         foreach(_FRAMEWORK {{ '${' }}{{ pkg_name }}_FRAMEWORKS_FOUND{{ config_suffix }}})
-            list(APPEND {{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} ${_FRAMEWORK})
             list(APPEND {{ pkg_name }}_LIBRARIES{{ config_suffix }} ${_FRAMEWORK})
         endforeach()
 
         foreach(_SYSTEM_LIB {{ '${' }}{{ pkg_name }}_SYSTEM_LIBS{{ config_suffix }}})
-            list(APPEND {{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} ${_SYSTEM_LIB})
             list(APPEND {{ pkg_name }}_LIBRARIES{{ config_suffix }} ${_SYSTEM_LIB})
         endforeach()
 
         # We need to add our requirements too
-        set({{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }} {{ '"${' }}{{ pkg_name }}_LIBRARIES_TARGETS{{ config_suffix }}{{ '};' }}{{ deps_targets_names }}")
-        set({{ pkg_name }}_LIBRARIES{{ config_suffix }} {{ '"${' }}{{ pkg_name }}_LIBRARIES{{ config_suffix }}{{ '};' }}{{ deps_targets_names }}")
+        set({{ pkg_name }}_LIBRARIES{{ config_suffix }} "")
```

Comment:
Hey, @lasote !

I think that here you have broken everything :) The variable was not cleared here before. It seems strange even when looking to several `"list(APPEND ...)"` above, which actively push something to this variable. 

I didn't understand all picture though. But the specific problem is: `conan=1.51` broke a build of `zstd` dependant, where I'm using `${zstd_LIBRARIES}`, which become empty after update. And I'm pretty sure that bug in this PR.

Reply:
Checking.